File: src/main/java/br/com/kronos/backend/aplicacao/resultado/Avaliado.java

package br.com.kronos.backend.aplicacao.resultado;

import static br.com.kronos.backend.aplicacao.validador.Validacoes.exigirNaoNulo;

import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;

@EqualsAndHashCode(of = "id")
@Getter
public class Avaliado {
    
    private Long id;
    private Long idAvaliacao;
    private Long idCredito;
    private Long idGrupoAvaliacao;
    private Double nota;
    private Long idMencao; 
    private boolean descarte; 
    private String motivoDescarte; 
  
	@Builder
    public Avaliado(Long id, Long idAvaliacao, Long idCredito, Long idGrupoAvaliacao, Double nota, 
                         Long idMencao, boolean descarte, String motivoDescarte  ) {

		super();
        this.id = id;
        this.idAvaliacao = exigirNaoNulo(idAvaliacao, "Avaliação"); 
        this.idCredito = exigirNaoNulo(idCredito, "Crédito (Aluno)");
        this.idGrupoAvaliacao = idGrupoAvaliacao; 
        this.nota = nota; 
        this.idMencao = idMencao; 
        this.descarte = exigirNaoNulo(descarte, "Descarte"); 
        this.motivoDescarte = motivoDescarte; 
	}
}

File: poc-kronos-frontend/src/model/avaliacao/Avaliado.js
class Avaliado {
  constructor(dados) {
    this.id = dados ? dados.id : null;
    this.idAvaliacao = dados ? dados.idAvaliacao : null;
    this.idCredito = dados ? dados.idCredito : null;
    this.idGrupoAvaliacao = dados ? dados.idGrupoAvaliacao : null;
    this.nota = dados ? dados.nota : null;
    this.idMencao = dados ? dados.idMencao : null;
    this.descarte = dados ? dados.descarte : null;
    this.motivoDescarte = dados ? dados.motivoDescarte : null;
    this.numeroRegistro = dados ? dados.numeroRegistro : null;
    this.nomeAvaliado = dados ? dados.nomeAvaliado : null;
    this.convertMencoes(dados ? dados.mencoes : []);
    this.situacao = dados ? dados.situacao : null;
  }

  convertMencoes(mencoes) {
    this.mencoes = "";
    mencoes.map(m => {
      this.mencoes += m + ", ";
    });
    this.mencoes = this.mencoes.substring(0, this.mencoes.length - 2);
  }

}
export default Avaliado;
